zio-http/src/main/scala/zio/http/Server.scala

```diff
@@ -34,45 +31,21 @@ object Server {
 
   val default = ServerConfig.live >>> live
 
-  val live: ZLayer[ServerConfig, Throwable, Server] = ZLayer.scoped {
+  val live = NettyDriver.default >>> base
+
+  val base: ZLayer[
+    Driver with AtomicReference[HttpApp[Any, Throwable]] with AtomicReference[Option[ErrorCallback]],
```

Comment:
This looks a bit ugly _and_ does not compile with Scala 3.2.0 (it somehow confuses the two `AtomicReference`s in `ZIO.service`. So I would wrap them in one or two case classes.

Reply:
Now that I read the rest of the PR: I think it would be best to expose these atomic references through `Driver` as they are created in `NettyDriver`'s layer anyway.

Reply:
I may be missing something here, but I am not sure how to get that to work since `ServerLive` needs both of those AtomicReferences as well.

Reply:
I think I am pretty close to what you are asking for, lemme know.

Reply:
Maybe I'm missing something but the new version seems a bit unnecessarily complicated to me - can't we just add those atomic references (or some abstraction over them) to `Driver` itself? Why do we have to expose `Driver & Driver.Context` in one layer and then require `Driver & Driver.Context` in another? 

If there is a good reason for the separation I'm fine with it but its not obvious to me.